Point_set_processing_3/include/CGAL/structure_point_set.h

```diff
@@ -0,0 +1,1483 @@
+            bool is_in_interval = false;
+            for (std::size_t i = 0; i < edge.indices.size(); ++ i)
+              {
+                std::size_t index_pt = edge.indices[i];
+                double dist = CGAL::squared_distance (m_corners[k].support,
+                                                      m_points[index_pt]);
+                if (m_status[index_pt] != SKIPPED
+                    && dist < 1.5 * d_DeltaEdge && dist > d_DeltaEdge / 2)
+                  {
+                    Vector move (m_corners[k].support,
+                                 m_points[index_pt]);
+                    if (move * m_corners[k].directions[ed] > 0.)
+                      {
+                        is_in_interval = true;
+                        break;
+                      }
+                  }
+              }
+
+            //rajouter un edge a 1 epsilon du cote dominant si pas de point entre SS_edge/2 et 3/2*SS_edge
+            if (!is_in_interval)
+              {
+                Point new_edge = m_corners[k].support + m_corners[k].directions[ed] * d_DeltaEdge;
+                m_points.push_back (new_edge);
+                m_normals.push_back (0.5 * m_planes[m_edges[m_corners[k].edges[ed]].planes[0]]->plane_normal()
+                                     + 0.5 * m_planes[m_edges[m_corners[k].edges[ed]].planes[1]]->plane_normal());
+                m_status.push_back (EDGE);
+                m_indices.push_back (m_corners[k].edges[ed]);
+                edge.indices.push_back (m_points.size() - 1);
+              }
+						
+            //rajouter un edge a 1/3 epsilon du cote dominant
+            Point new_edge = m_corners[k].support + m_corners[k].directions[ed] * d_DeltaEdge / 3;
+            m_points.push_back (new_edge);
+            m_normals.push_back (0.5 * m_planes[m_edges[m_corners[k].edges[ed]].planes[0]]->plane_normal()
+                                 + 0.5 * m_planes[m_edges[m_corners[k].edges[ed]].planes[1]]->plane_normal());
+            m_status.push_back (EDGE);
+            m_indices.push_back (m_corners[k].edges[ed]);
+            edge.indices.push_back (m_points.size() - 1);
+          }
+      }
+
+  }
+  /// \endcond    
+};
+
+
+  
+
+
+// ----------------------------------------------------------------------------
+// Public section
+// ----------------------------------------------------------------------------
+
+/// \ingroup PkgPointSetProcessing
+  
+/// This is an implementation of the Point Set Structuring algorithm. This
+/// algorithm takes advantage of a set of detected planes: it detects adjacency
+/// relationships between planes and resamples the detected planes, edges and
+/// corners to produce a structured point set.
+///
+/// The size parameter `epsilon` is used both for detecting adjacencies and for
+/// setting the sampling density of the structured point set.
+///
+/// For more details, please refer to \cgalCite{cgal:la-srpss-13}.
+///
+/// @tparam InputIterator Iterator over input points
```

Comment:
IIUC, I'm not sure `InputIterator` is any iterator over points but must be `Traits::Input_iterator`